src/server/scripts/Spells/spell_monk.cpp

```diff
@@ -164,6 +166,36 @@ class spell_monk_provoke : public SpellScript
     }
 };
 
+// 109132 - Roll
+class spell_monk_roll : public SpellScript
+{
+    PrepareSpellScript(spell_monk_roll);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MONK_ROLL_TRIGGER, SPELL_MONK_ROLL_FEATHER_FALL });
+    }
+
+    SpellCastResult CheckCast()
+    {
+        if (GetCaster()->HasUnitState(UNIT_STATE_ROOT))
+            return SPELL_FAILED_ROOTED;
+        return SPELL_CAST_OK;
+    }
+
+    void HandleDummy(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetCaster(), SPELL_MONK_ROLL_TRIGGER, true);
```

Comment:
does the sniff contain a smsg_spell_start for the triggered effects? If yes, remove the true

Reply:
ye, so no need to cast it as triggered = true then. Do normal casts instead. triggered = true skips the spell_start part coreside and goes straight for spell_go

Reply:
oh ty so only GetCaster()->CastSpell(GetCaster(), SPELL_MONK_ROLL_TRIGGER) ? idk what is happening with this pr it doesnt detect changes or compiles XD

Reply:
I recommend triple checking that the script still works and spell casts properly before discarding that triggered argument

Reply:
Have you tried rolling backwards? Just curious